src/Microsoft.TemplateEngine.Edge/Template/TemplateCreator.cs

```diff
@@ -457,6 +463,11 @@ private bool CheckForMissingRequiredParameters(InputDataSet templateParams, IPar
             InputDataSet evaluatedParameterSetData = parametersBuilder.Build(!isEvaluatedExternally, template.Generator, _logger);
             List<string> defaultParamsWithInvalidValues = new List<string>();
 
+            // Remove parameters from template - to prevent their usage in generators
+            evaluatedParameterSetData.Values
```

Comment:
imo this should be up to generator itself. `TemplateCreator` should just tell which parameters are disabled.